examples/step-75/step-75.cc

```diff
@@ -0,0 +1,1502 @@
+
+    {
+      pcout << "Calculating transformation matrices..." << std::endl;
+      TimerOutput::Scope t(computing_timer, "calculate transformation");
+      legendre->precalculate_all_transformation_matrices();
+    }
+
+    for (unsigned int cycle = 0; cycle < prm.n_cycles; ++cycle)
+      {
+        pcout << "Cycle " << cycle << ':' << std::endl;
+
+        if (cycle == 0)
+          initialize_grid();
+        else
+          adapt_resolution();
+
+        setup_system();
+
+        print_diagnostics();
+
+        solve_system();
+
+        compute_indicators();
+
+        if (Utilities::MPI::n_mpi_processes(mpi_communicator) <= 32)
+          output_results(cycle);
+
+        computing_timer.print_summary();
+        computing_timer.reset();
+
+        pcout << std::endl;
+      }
+  }
+} // namespace Step75
+
+
+
+// @sect4{main()}
+
+// The final function is the <code>main</code> function that will ultimately
+// create and run a LaplceOperator instantiation. Its structure is similar to
+// most other tutorial programs. We define all parameters used in the tutorial
+// here, so this is a good opportunity to justify how we picked them.
+//
+// We provide ranges for levels on which the function space is operating with a
+// reasonable resolution. The minimal polynomial degree is set to 2 as it seems
+// that the smoothness estimation algorithms have trouble with linear elements,
+// so we will skip them altogether.
+//
+// We will use general refine and coarsen fractions that have been elaborated in
+// the other deal.II tutorials: using the fixed number strategy, we will flag
+// 30% of all cells for refinement and 3% for coarsening. For the p-adaptation
+// fractions, we will take an educated guess. Since we only expect a single
+// singularity in our scenario, i.e., in the origin of the domain, and a smooth
+// solution anywhere else, we would like to strongly perfer to use p-adaptation
+// over h-adaptation. This reflects in our choice of a fraction of 90% for both
+// p-refinement and p-coarsening.
+//
+// For load balancing, efficient solvers like the one we use should scale
+// linearly with the number of degrees of freedom owned. We will set the
+// parameters for cell weighting correspondigly.
+int main(int argc, char *argv[])
+{
+  try
+    {
+      using namespace dealii;
+      using namespace Step75;
+
+      Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+
+      Parameters prm;
+      {
+        prm.n_cycles         = 8;
+        prm.tolerance_factor = 1e-12;
+
+        prm.min_h_level            = 5;
+        prm.max_h_level            = 12;
+        prm.min_p_degree           = 2;
+        prm.max_p_degree           = 6;
+        prm.max_p_level_difference = 1;
+
+        prm.refine_fraction    = 0.3;
+        prm.coarsen_fraction   = 0.03;
+        prm.p_refine_fraction  = 0.9;
+        prm.p_coarsen_fraction = 0.9;
+
+        prm.weighting_factor   = 1e6;
+        prm.weighting_exponent = 1.;
```

Comment:
Why not make these parameters default? Just like in the MG case?

Reply:
We could do that to reduce a few lines.

The reason I put it here was to justify the choice of parameters after all the theoretical details have been covered in the in-source documentation. But we could also move the justification to the places where they are actually used.